src/forms/StripeConnectAccountForm/StripeConnectAccountForm.js

```diff
@@ -0,0 +1,278 @@
+import React, { useState } from 'react';
+import { bool, func, object, shape, string } from 'prop-types';
+import { compose } from 'redux';
+import { FormattedMessage, injectIntl, intlShape } from '../../util/reactIntl';
+import { Form as FinalForm } from 'react-final-form';
+import arrayMutators from 'final-form-arrays';
+import classNames from 'classnames';
+import config from '../../config';
+import { isStripeError } from '../../util/errors';
+import * as validators from '../../util/validators';
+import {
+  Button,
+  InlineTextButton,
+  FieldSelect,
+  Form,
+  StripeBankAccountTokenInputField,
+} from '../../components';
+
+import css from './StripeConnectAccountForm.css';
+
+const supportedCountries = config.stripe.supportedCountries.map(c => c.code);
+
+export const stripeCountryConfigs = countryCode => {
+  const country = config.stripe.supportedCountries.find(c => c.code === countryCode);
+
+  if (!country) {
+    throw new Error(`Country code not found in Stripe config ${countryCode}`);
+  }
+  return country;
+};
+
+const countryCurrency = countryCode => {
+  const country = stripeCountryConfigs(countryCode);
+  return country.currency;
+};
+
+const CreateStripeAccountFields = props => {
+  const { disabled, countryLabel, values, intl } = props;
+  const country = values.country;
+  const countryRequired = validators.required(
+    intl.formatMessage({
+      id: 'StripeConnectAccountForm.countryRequired',
+    })
+  );
+  return (
+    <div className={css.sectionContainer}>
+      <FieldSelect
+        id="country"
+        name="country"
+        disabled={disabled}
+        className={css.selectCountry}
+        autoComplete="country"
+        label={countryLabel}
+        validate={countryRequired}
+      >
+        <option disabled value="">
+          {intl.formatMessage({ id: 'StripeConnectAccountForm.countryPlaceholder' })}
+        </option>
+        {supportedCountries.map(c => (
+          <option key={c} value={c}>
+            {intl.formatMessage({ id: `StripeConnectAccountForm.countryNames.${c}` })}
+          </option>
+        ))}
+      </FieldSelect>
+
+      {country ? (
+        <StripeBankAccountTokenInputField
+          className={css.bankDetailsStripeField}
+          disabled={disabled}
+          name="bankAccountToken"
+          formName="StripeConnectAccountForm"
+          country={country}
+          currency={countryCurrency(country)}
+          validate={validators.required(' ')}
+        />
```

Comment:
Is this Field used just for visuals or is it editable? I don't understand why there's a dropdown and an input with the value of the selected option from the dropdown.

Reply:
This StripeBankAccountTokenInputField is used for getting the bank account token from Stripe. It's the place where we collect the bank account number (e.g. IBAN). Because it depends on the country which fields should be shown (IBAN has only one field but e.g. Canadian bank account has three different fields) we need to ask the country first. That's why the selected option is passed to input. Also, country and currency are required information for generating the bank account token.